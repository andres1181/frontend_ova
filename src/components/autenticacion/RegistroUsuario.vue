<script>

	export default {
		name: 'RegistroUsuario',

		data() {
			return {
				info: {
					nomUsuario: '',
					codigo: '',
					email: '',
					tipo: '',
					contrasena: '',
					contrasena2: ''
				},
				rules: {
					usuarioRules: [v => !!v || 'Campo requerido'],
					codigoRules: '',
					emailRules: '',
					tipoRules: '',
					contrasenaRules: '',
				}
			}
		}
	}

</script>

<template>

 <v-form id="formRegistro">

		<v-text-field :rules="rules.usuarioRules" id="nomUsuario">
		</v-text-field>
		 <v-text-field :rules="rules.codigoRules" id="codUsuario">
		</v-text-field>
		<v-text-field :rules="rules.emailRules" id="emailUsuario">
		</v-text-field>
		<v-radio-group :rules="rules.tipoRules" id="tipoUsuario">
			<v-radio  id="radioDocente"></v-radio>
			<v-radio  id="radioEstudiante"></v-radio>
		</v-radio-group>
		<v-text-field :rules="rules.contrasenaRules" id="contrasena">
		</v-text-field>
		<v-text-field :rules="rules.contrasenaRules" id="repertirContrasena">
		</v-text-field>
	 </v-form>

</template>
