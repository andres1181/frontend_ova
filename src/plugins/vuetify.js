import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify,
  { //estilos de otras cosasâ€¦
    theme: {
    themes: {

      light: {
        main: '#3cd3ad',
        primary: '#3cd3ad',
        secondary: '#424242',
        accent: '#82B1FF',
        error: '#FF5252',
        info: '#2196F3',
        success: '#4CAF50',
        warning: '#FFC107',
      },
      dark: {
        primary: '#4cb8c4',
      },
    },
  },
});

export default new Vuetify({
});
