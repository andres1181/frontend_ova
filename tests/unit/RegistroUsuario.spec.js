import {
  shallowMount,
  mount,
  createLocalVue
} from '@vue/test-utils'
import vuetify from "vuetify"
import Vue from 'vue';
import RegistroUsuario from '@/components/autenticacion/RegistroUsuario.vue'


describe('RegistroUsuario', () => {
  let wrapper;
  beforeEach(() => {
    Vue.use(vuetify)

    wrapper = mount(RegistroUsuario)
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it("Existe un elemento  llamado <<formRegistro>>", () => {
    const getElementById = wrapper.find('#formRegistro')
    expect(getElementById.element.id).toBe('formRegistro')
  })

  const data = RegistroUsuario.data()
  it("Los datos se inicializan correctamente", () => {

    expect(data.info.nomUsuario).toBe('')
    expect(data.info.codigo).toBe('')
    expect(data.info.email).toBe('')
    expect(data.info.tipo).toBe('')
    expect(data.info.contrasena).toBe('')
    expect(data.info.contrasena2).toBe('')

  })


  it("Existe un elemento llamado <<nomUsuario>>", () => {
    const getElementById = wrapper.find('#nomUsuario')
    expect(getElementById.element.id).toBe('nomUsuario')
  })
  it("Existe un elemento llamado <<codUsuario>>", () => {
    const getElementById = wrapper.find('#codUsuario')
    expect(getElementById.element.id).toBe('codUsuario')
  })
  it("Existe un elemento llamado <<emailUsuario>>", () => {
    const getElementById = wrapper.find('#emailUsuario')
    expect(getElementById.element.id).toBe('emailUsuario')
  })
  it("Existen un elemento para el tipo de usuario llamado radioDocente", () => {
    const getElementById = wrapper.find('#radioDocente')
    expect(getElementById.element.id).toBe('radioDocente')
  })
  it("Existen un elemento para el tipo de usuario llamado radioEstudiante", () => {
    const getElementById = wrapper.find('#radioEstudiante')
    expect(getElementById.element.id).toBe('radioEstudiante')
  })
  it("Existen un elemento  llamado <<tipoUsuario>>", () => {
    const getElementById = wrapper.find('#tipoUsuario')
    expect(getElementById.element.id).toBe('tipoUsuario')
  })
  it("Existe un elemento llamado <<contrasena>>", () => {
    const getElementById = wrapper.find('#contrasena')
    expect(getElementById.element.id).toBe('contrasena')
  })
  it("Existe un elemento llamado <<repertirContrasena>>", () => {
    const getElementById = wrapper.find('#repertirContrasena')
    expect(getElementById.element.id).toBe('repertirContrasena')
  })

  it("Existen variables para almacenar las reglas de los campos", () => {

    expect(data.rules.usuarioRules).toBe('[v => !!v || 'Campo requerido']')
    const barByName = wrapper.findComponent({ name: 'bar' }) // => finds Bar by `name`
expect(barByName.exists()).toBe(true)

  })


  /*expect(data.rules.codigoRules).exists()
  expect(data.rules.emailRules).exists()
  expect(data.rules.tipoRules).exists()
  expect(data.rules.contrasenaRules).exists()*/

  // it("Tiene un boton llamado registrar", () => {
  //   const getElementById = wrapper.find('#registrar')
  //   expect(getElementById.element.id).toBe('registrar')
  // })
})
//expect(wrapper.find('.regis').exists())
