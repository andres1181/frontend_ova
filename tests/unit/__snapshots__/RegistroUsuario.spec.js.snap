// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistroUsuario renders correctly 1`] = `
<form
  class="v-form"
  id="formRegistro"
  novalidate="novalidate"
>
  <div
    class="v-input theme--light v-text-field"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <input
            id="nomUsuario"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="v-input theme--light v-text-field"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <input
            id="codUsuario"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="v-input theme--light v-text-field"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <input
            id="emailUsuario"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
        style="height: auto;"
      >
        <div
          aria-labelledby="tipoUsuario"
          class="v-input--radio-group__input"
          id="tipoUsuario"
          role="radiogroup"
        >
          <div
            class="v-radio theme--light"
          >
            <div
              class="v-input--selection-controls__input"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate material-icons theme--light"
              >
                $radioOff
              </i>
              <input
                aria-checked="false"
                id="radioDocente"
                name="radio-37"
                role="radio"
                type="radio"
                value=""
              />
              <div
                class="v-input--selection-controls__ripple"
              />
            </div>
          </div>
           
          <div
            class="v-radio theme--light"
          >
            <div
              class="v-input--selection-controls__input"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate material-icons theme--light"
              >
                $radioOff
              </i>
              <input
                aria-checked="false"
                id="radioEstudiante"
                name="radio-37"
                role="radio"
                type="radio"
                value=""
              />
              <div
                class="v-input--selection-controls__ripple"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="v-messages theme--light"
      >
        <transition-group-stub
          class="v-messages__wrapper"
          name="message-transition"
          tag="div"
        />
      </div>
    </div>
  </div>
   
  <div
    class="v-input theme--light v-text-field"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <input
            id="contrasena"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="v-input theme--light v-text-field"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <input
            id="repertirContrasena"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
</form>
`;
